app/src/main/java/adapter/SoundAdapter.kt

```diff
@@ -0,0 +1,102 @@
+
+    companion object {
+        const val TAG = "Youpidapp::SoundAdapter"
+    }
+
+    private val soundsList: MutableList<Sound> = _SoundsList.toMutableList()
+    private val activeViewHolderForSound = HashMap<Sound, SoundViewHolder>()
+
+    interface SoundClickedListener {
+        fun onSoundClicked(soundViewHolder: SoundViewHolder)
+    }
+
+    /** Used for informing the front-end that the button was unbound, and thus, its properties should be restored to the initial state. */
+    interface CleanupAnimationListener {
+        fun onUnbind(soundViewHolder: SoundViewHolder)
+    }
+
+    class SoundViewHolder(itemView: View, soundClickedListener: SoundClickedListener) : RecyclerView.ViewHolder(itemView) {
+
+        val button: Button = itemView.findViewById(R.id.sound_button_id)
+        var sound: Sound? = null  // TODO replace by a call to the adapter, like the following; may fail because I don't know what is the correct position to use: adapter or layout
+        /*lateinit var adapter: SoundAdapter
+        val sound: Sound?
+            get() {
+                return adapter.soundsList[adapterPosition]
+            }*/
+        var cleanupAnimationListener: CleanupAnimationListener? = null
+
+        init {
+            button.setOnClickListener {
+                soundClickedListener.onSoundClicked(this)
+            }
+        }
+
+        fun bind(sound: Sound, adapter: SoundAdapter) {
+            this.sound = sound
+            button.text = sound.displayText
+            adapter.activeViewHolderForSound[sound] = this
+            Log.d(TAG, "VH $this bind called ")
+        }
+
+        fun unbind(adapter: SoundAdapter) {
+            adapter.activeViewHolderForSound.remove(sound)
+            cleanupAnimationListener?.onUnbind(this)
+            Log.d(TAG, "VH $this unbind called")
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SoundViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.sound_button, parent, false)
+        return SoundViewHolder(view, soundClickedListener)
+    }
+
+    override fun onBindViewHolder(holder: SoundViewHolder, position: Int) {
+        holder.bind(soundsList[position], this)
+        Log.d(TAG, "VH bound $holder ${holder.button}")
+    }
+
+    override fun onViewRecycled(holder: SoundViewHolder) {
+        super.onViewRecycled(holder)
+        holder.unbind(this)
+        Log.d(TAG, "VH recycled $holder")
+    }
+
+    fun getActiveViewHolder(sound: Sound): SoundViewHolder? {
+        return activeViewHolderForSound[sound]
+    }
+
+    fun add(sound: Sound) {
+        soundsList.add(sound)
+    }
+
+    fun add(sounds: List<Sound>) {
+        soundsList.addAll(sounds)
+    }
+
+    fun remove(sound: Sound) {
+        soundsList.remove(sound)
+    }
+
+    fun remove(sounds: List<Sound>) {
+        soundsList.removeAll(sounds)  // Maybe needs to use another method
+    }
```

Comment:
Are those functions really needed? They do not seem to be called anywhere...

Reply:
They will be used for the search feature, but yes, they are unused.